spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/util/ArgumentSanitizer.java

```diff
@@ -204,4 +214,75 @@ public HttpHeaders sanitizeHeaders(HttpHeaders headers) {
         }
         return result;
     }
+	/**
+	 * Will replace sensitive string value in the Map with '*****'
+	 *
+	 * @param input to be sanitized
+	 * @return the sanitized map.
+	 */
+	public Map<String, Object> sanitizeMap(Map<String, Object> input) {
+		Map<String, Object> result = new HashMap<>();
+		for (Map.Entry<String, Object> entry : input.entrySet()) {
+			if (entry.getValue() instanceof String) {
+				result.put(entry.getKey(), sanitize(entry.getKey(), (String) entry.getValue()));
+			} else if (entry.getValue() instanceof Map) {
+				Map<String, Object> map = (Map<String, Object>) entry.getValue();
+				result.put(entry.getKey(), sanitizeMap(map));
+			} else {
+				result.put(entry.getKey(), entry.getValue());
+			}
+		}
+		return result;
+	}
+
+	/**
+	 * Will replace the sensitive string fields with '*****'
+	 *
+	 * @param input to be sanitized
+	 * @return The sanitized JSON string
+	 * @throws JsonProcessingException
+	 */
+	public String sanitizeJsonString(String input) throws JsonProcessingException {
+		ObjectMapper mapper = new ObjectMapper();
+		Map<String, Object> data = mapper.readValue(input, new TypeReference<Map<String, Object>>() {
+		});
+		return mapper.writeValueAsString(sanitizeMap(data));
+	}
+
+	/**
+	 * Will replace the sensitive string fields with '*****'
+	 *
+	 * @param input to be sanitized
+	 * @return The sanitized YAML string
+	 * @throws JsonProcessingException
+	 */
+	public String sanitizeYamlString(String input) throws JsonProcessingException {
+		ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+		Map<String, Object> data = mapper.readValue(input, new TypeReference<Map<String, Object>>() {});
+		return mapper.writeValueAsString(sanitizeMap(data));
+	}
+	/**
+	 * Will determine the type of data and treat as JSON or YAML to sanitize sensitive values.
```

Comment:
Do you mind adding a callout as to what the "indicator" is that it used to decide?